### PR TITLE
Enable linux-hosted build and tests run

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
@cijothomas . Unit tests related to Acl are still failing when run on Linux. Maybe they need to be marked as windows only.

I enabled Linux build using yml definition. Check out build run results: https://mseng.visualstudio.com/AppInsights/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=6237 Once unit tests fixed - this build can be enabled for CI tests

### CODE REVIEW

File: .vsts/linux-build.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  inputs:
+    command: "test"
+    projects: "Test/Microsoft.ApplicationInsights.Test/netcoreapp11/Microsoft.ApplicationInsights.netcoreapp11.Tests.csproj"
+    arguments: "--configuration Release"
+
+- task: DotNetCoreCLI@1
+  inputs:
+    command: "test"
+    projects: "Test/ServerTelemetryChannel.Test/NetCore.Tests/TelemetryChannel.NetCore.Tests.csproj"
+    arguments: "--configuration Release"
```
Comment (cijothomas) [inline .vsts/linux-build.yml]: displayName: Run tests for TelemetryChannel

**Comment by cijothomas** [comment]
@SergeyKanzhelev Yes, the test code itself contains 'Windows Specific' code. So they need to be somehow excluded when run in non-windows.

**Comment by d3r3kk** [comment]
@cijothomas and @SergeyKanzhelev can we make use of Category decorations on our test methods?

Something like:

    [TestMethod]
    [TestCategory("Linux")]
    public void TestSomethingOutOnLinuxOnly()
    {
        Assert.AreEqual(System.Runtime.InteropServices.RuntimeInformation.OSDescription,"Linux");
    }

    [TestMethod]
    [TestCategory("Windows")]
    public void TestSomethingOutOnWindowsOnly()
    {
        Assert.AreEqual(System.Runtime.InteropServices.RuntimeInformation.OSDescription,"Windows");
    }

and then in our Test runs for Linux, run only those tests with `--filter TestCategory!=WindowsOnly`, and on our test runs on Windows boxes,  use `--filter TestCategory!=LinuxOnly`

See [here for filtering rules using vstest](https://github.com/Microsoft/vstest-docs/blob/master/docs/filter.md), and [here for an overview of other test frameworks](https://docs.microsoft.com/en-us/dotnet/core/testing/).

**Comment by SergeyKanzhelev** [comment]
@d3r3kk feel free to try it out. Just update yml file and add category attributes. And then run mentioned build on mseng. I also might get to it later

**Comment by SergeyKanzhelev** [comment]
@d3r3kk nevermind. You suggestion worked like a charm

**Pull request merged**